python/ray/air/execution/resources/resource_manager.py

```diff
@@ -0,0 +1,47 @@
+import abc
+
+from typing import List, Optional
+
+import ray
+from ray.air.execution.resources.request import (
+    ResourceRequest,
+    AllocatedResource,
+)
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class ResourceManager(abc.ABC):
```

Comment:
IMO, this interface is very tailored towards the limitations of placement group interface and not generic enough. Should `get_source_futures` and `update_state` etc be moved down to its PlacementGroupResourceManager?

Reply:
Both are needed because Core doesn't provide any events. We need to keep the interface though to be able to plug in different resource managers.

The default for the resource futures is to just return an empty list, and update state is a no-op. I'll add documentation to point out the contract with these methods.

Just for context, calling these methods is not *required*, it's just a convenience to be able to await different futures (e.g. training and resource futures) like we do in ray tune right now. As an alternative users can just do

```
while not resource_manager.has_resources_ready(request):
    time.sleep(0.1)
```

(but we don't want to do this in Ray Tune)

Reply:
As discussed offline, let's leave it with the dummy interface. If we ever get to the point where we have an event-based manager we can just remove this interface.